### PR TITLE
feat(agent): persist session event log

### LINKED ISSUE #2050: Agent session activity is durably recorded for replay
**Type:** Feature

Part of the detachable-agent-sessions epic (#955).

## What

Agent session activity today lives only in memory plus a debounced JSON snapshot of current state (`MingaAgent.SessionStore`). There is no durable, ordered record of *what happened and when*. That means a client that was disconnected has no reliable way to ask "what did I miss," and a crashed BEAM leaves no audit trail of in-flight work. We need every agent session to durably record its activity as an append-only, ordered event log.

## Why

The durable event log is the foundation for detach/reattach (#955). Current-state snapshots cannot answer "replay everything after the last thing I saw," which is exactly what a reconnecting client needs. The same log gives us a crash/debug audit trail and a way to mark interrupted work after a process crash. Without it, every later piece of the epic (reconnect catch-up, multi-client attach) has to invent its own ad-hoc history. Build the source of truth once.

## Acceptance Criteria

1. ✅ Every running agent session durably records its activity to an append-only store that survives a full BEAM restart. Addressed by #2064.
   - Each recorded entry has a monotonically increasing id, the owning session id (the daemon-stable id from #2053, not the in-memory `session-N`), an event type, a payload, and a wall-clock timestamp.
   - Records are never mutated or reordered after they are written.
2. ✅ A reader can fetch "all events for session X with id greater than N, in order" without blocking the writer. Addressed by #2064.
   - Asking with N = the latest id returns an empty list.
   - Asking with N = 0 returns the full ordered history for that session.
3. ✅ Recording an event never blocks or slows down the agent session producing it (writes are asynchronous to the session process). Addressed by #2064.
4. ✅ The recorded event taxonomy covers the activity a client needs to reconstruct a session view: session start/stop, assistant output, tool call start/finish, file edits proposed, approval requested/resolved, system messages, and waiting-for-input. Addressed by #2064.
5. ✅ The log works in headless mode (a `minga --headless` server with no GUI records session activity the same way). Addressed by #2064.
6. ✅ History is queryable after a simulated crash: killing and restarting the session process, then reading the log, returns the events produced before the crash. Addressed by #2064.
7. ✅ The log never contains secrets: provider API keys and credential/config values do not appear in any recorded event payload, inheriting the `MingaAgent.Credentials` rule that keys are never logged or exported. The same redaction holds for any snapshot built from the log. Addressed by #2064.
8. ✅ Recorded payloads are replay-safe: they contain no process identifiers or other node-local references, only JSON-serializable data. A tool approval is recorded with a stable approval id (not a `reply_to` PID), so a client that reconnects later can act on an approval that was requested while it was away (see #2051). Addressed by #2064.

### PR DESCRIPTION
**Pull request by jsmestad**
# TL;DR
Agent sessions now write a durable append-only event log for replay. The log records stable session events asynchronously, supports cursor reads by session, and redacts credential-like payload keys before SQLite persistence.

Closes #2050

## Context
This is the next detachable-agent-sessions slice after #2053. It gives reconnect and replay work a durable source of truth, so a detached client can later ask for events after its last seen id instead of relying on in-memory state.

## Changes
- Added `MingaAgent.EventLog`, a daemon-level GenServer writer supervised under `Minga.Services.Independent` so it runs in headless mode.
- Added an agent-specific SQLite store with WAL mode, append-only rows, schema versioning, retention sweep support, and `events_after/4` cursor reads.
- Added `EventRecord` and `Taxonomy` modules for the durable event shape and canonical event names.
- Wired `MingaAgent.Session` to record session start/stop, user prompts, assistant deltas, tool lifecycle, file edits, approvals, system messages, status changes, waiting-for-input, context, errors, and driver changes.
- Sanitized persisted payloads so credential-like keys, including camelCase and hyphenated variants, are redacted; PIDs and references are replaced with replay-safe placeholders.
- Added event-log tests for cursor ordering, durability after reopen, async recording, redaction, session broadcast recording, steering prompt replay, and history after a simulated session crash.

## Verification
- `mix compile --warnings-as-errors`
- `mix test.debug test/minga_agent/event_log_test.exs test/minga_agent/event_log/store_test.exs test/minga_agent/event_log/session_integration_test.exs`
- `mix test.llm test/minga_agent/event_log_test.exs test/minga_agent/event_log test/minga_agent/session_test.exs`
- `make lint`
- `mix test.llm` (56 doctests, 91 properties, 9098 tests, 0 failures, 292 excluded)
- `git diff --check`
- Bug-hunt rechecks: PASS
- Final reviewer gate: PASS

## Acceptance Criteria Addressed
- Running sessions durably record append-only activity with monotonic ids, stable session ids, event type, payload, wall-clock timestamp, and monotonic timestamp ✅
- Readers can fetch events for a session with `id > N` in order without blocking the writer ✅
- Recording is asynchronous to the session process ✅
- The persisted taxonomy covers the session activity needed to reconstruct a session view ✅
- The log runs in headless mode through `Minga.Services.Independent` ✅
- History remains queryable after a simulated session crash ✅
- Credential-like values are redacted before persistence ✅
- Payloads are JSON/replay-safe and avoid process-local references ✅

### CODE REVIEW

File: lib/minga_agent/session.ex [hunk 1]
```diff
@@ -2451,6 +2596,13 @@ defmodule MingaAgent.Session do
 
   @impl GenServer
   def terminate(reason, state) do
+    EventLog.record(
+      state.session_id,
+      :session_stopped,
+      %{reason: inspect(reason), status: state.status},
+      state.event_log_server
+    )
```
[the remark this answers is not in the document]
Comment (jsmestad) [reply lib/minga_agent/session.ex:2627]: Not a concern in practice for two reasons:

1. **Shutdown ordering protects this path.** Sessions run under `MingaAgent.Supervisor`, while `EventLog` runs under `Minga.Services.Independent`. During orderly application shutdown, sessions terminate first and their `terminate/2` callbacks fire the cast. The `Independent` supervisor (and EventLog) is still alive to process it.

2. **`:session_stopped` is explicitly best-effort.** A `Process.exit(pid, :kill)` bypasses `terminate/2` entirely, so there is already no guarantee this event is recorded. The test suite acknowledges this by not asserting `:session_stopped` after a kill. Making this path synchronous would block session shutdown for SQLite I/O, which is a worse tradeoff than the rare edge case of losing a stop event during a race in abnormal shutdown.

No change needed.

File: lib/minga_agent/session.ex [hunk 2]
```diff
@@ -647,6 +650,17 @@ defmodule MingaAgent.Session do
       credentials_configured: true
     }
 
+    EventLog.record(
+      state.session_id,
+      :session_started,
```
[the remark this answers is not in the document]
Comment (jsmestad) [reply lib/minga_agent/session.ex:655]: Good catch. Fixed in the latest push: `handle_call(:new_session, ...)` now records `:session_stopped` for the old session_id before the swap and `:session_started` for the new one after.

File: lib/minga_agent/session.ex [hunk 3]
```diff
@@ -1867,13 +1915,90 @@ defmodule MingaAgent.Session do
...
+            tool_name: tool_name
+          }}
+
+  defp event_log_entry({:approval_pending, approval}),
+    do: {:approval_requested, Map.put(approval, :approval_id, approval.tool_call_id)}
+
+  defp event_log_entry({:approval_resolved, _decision}), do: nil
+
+  defp event_log_entry({:system_message, message, level}),
+    do: {:system_message, %{message: message, level: level}}
```
[the remark this answers is not in the document]
Comment (jsmestad) [reply lib/minga_agent/session.ex]: Valid. Fixed by recording `:system_message` directly inside `append_system_message/3`, which is the single point all system messages flow through. The broadcast-path clause for `{:system_message, ...}` now returns `nil` to prevent duplicate recording.

File: lib/minga_agent/event_log.ex [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+  def terminate(_reason, state) do
+    Store.close(state.db)
+    :ok
+  end
+
+  @spec open_or_recreate(String.t()) :: {:ok, Store.db()} | {:error, term()}
+  defp open_or_recreate(path) do
+    case Store.open(path) do
+      {:ok, db} -> {:ok, db}
+      {:error, _reason} -> recreate_database(path)
```
[the remark this answers is not in the document]
Comment (jsmestad) [reply lib/minga_agent/event_log.ex]: Good catch. Fixed: `open_or_recreate` now runs a quick integrity check before deleting. Only a confirmed-corrupt database gets recreated. Transient failures (lock contention, I/O pressure, permissions) propagate to the supervisor for retry, preserving existing history.

File: lib/minga_agent/session.ex [hunk 5]
```diff
@@ -1867,13 +1940,89 @@ defmodule MingaAgent.Session do
...
+            tool_call_id: tool_call_id,
+            tool_name: tool_name
+          }}
+
+  defp event_log_entry({:approval_pending, approval}),
+    do: {:approval_requested, Map.put(approval, :approval_id, approval.tool_call_id)}
+
+  defp event_log_entry({:approval_resolved, _decision}), do: nil
+
+  defp event_log_entry({:system_message, _message, _level}), do: nil
```
[the remark this answers is not in the document]
Comment (jsmestad) [reply lib/minga_agent/session.ex:1993]: Good find. Fixed: `maybe_show_auth_onboarding` now routes through `append_system_message` (which records to the event log) instead of calling `append_msg` directly. The explicit `broadcast({:system_message, ...})` is kept for subscriber UI notification, and `event_log_entry({:system_message, ...})` returns `nil` so there's no double recording.
